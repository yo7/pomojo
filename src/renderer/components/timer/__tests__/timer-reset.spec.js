import Vue from 'vue'
import Vuex from 'vuex'
import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import TimerReset from '../timer-reset.vue'

Vue.use(Vuex)

describe('TimerReset', () => {
  let store

  it('shows reset text if resettable', () => {
    const getters = {'timer/resettable': () => true}
    store = new Vuex.Store({getters})

    const wrapper = shallow(TimerReset, {store})
    const text = wrapper.find('.timer-reset > span')
    expect(text.hasStyle('display', 'none')).toBeFalsy()
  })

  it('does not show reset text unless resettable', () => {
    const getters = {'timer/resettable': () => false}
    store = new Vuex.Store({getters})

    const wrapper = shallow(TimerReset, {store})
    const text = wrapper.find('.timer-reset > span')
    expect(text.hasStyle('display', 'none')).toBeTruthy()
  })

  it('reset timer on click', () => {
    const getters = {'timer/resettable': () => true}
    const actions = {
      'timer/reset': jest.fn()
    }
    store = new Vuex.Store({getters, actions})
    const wrapper = shallow(TimerReset, {store})
    const text = wrapper.find('.timer-reset > span')
    text.trigger('click')
    expect(actions['timer/reset']).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    const getters = {'timer/resettable': () => true}
    store = new Vuex.Store({getters})

    const wrapper = shallow(TimerReset, {store})
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
