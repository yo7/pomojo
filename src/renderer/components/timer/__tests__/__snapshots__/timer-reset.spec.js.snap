// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerReset if resettable renders correctly 1`] = `
<div class="timer-reset"><span style="display:none;">
    reset
  </span></div>
`;

exports[`TimerReset unless resettable renders correctly 1`] = `
<div class="timer-reset"><span style="display:none;">
    reset
  </span></div>
`;
